### PR TITLE
Provide better error message when adding unnamed attrib values

### PR DESCRIPTION
**Pull request by schloerke**
Fixes https://github.com/rstudio/htmltools/issues/222

### CODE REVIEW

File: R/tags.R [hunk 1]
```diff
@@ -292,7 +292,11 @@ tagFunction <- function(func) {
 #' @export
 tagAppendAttributes <- function(tag, ...) {
   throw_if_tag_function(tag)
-  tag$attribs <- c(tag$attribs, dropNullsOrEmpty(dots_list(...)))
+  newAttribs <- dropNullsOrEmpty(dots_list(...))
+  if (any(names2(newAttribs) == "")) {
+    warning("At least one of the new attribute values did not have a name. Did you forget to include an attribute name?")
```
Comment (cpsievert) [inline R/tags.R:297]: I feel like we should `stop()` here since the error you'll eventually get isn't useful

File: R/tags.R [hunk 2]
```diff
@@ -492,6 +496,8 @@ tagWrite <- function(tag, textWriter, indent=0, eol = "\n") {
 
   # write attributes
   for (attrib in names(attribs)) {
+    # Can not display attrib without a key
+    if (identical(attrib, "")) next
```
Comment (cpsievert) [inline R/tags.R:500]: Oh, I see, so this would 'fix' this situation?

```r
x <- div()
x$attribs[[1]] <- "a"
```

Comment (cpsievert) [reply R/tags.R:500]: Part of me thinks that we should also `stop()` here with an informative error

Comment (schloerke) [reply R/tags.R:500]: Correct. No error is thrown.

Tested below

File: R/tags.R [hunk 3]
```diff
@@ -492,6 +499,13 @@ tagWrite <- function(tag, textWriter, indent=0, eol = "\n") {
 
   # write attributes
   for (attrib in names(attribs)) {
+    # Can not display attrib without a key
+    if (identical(attrib, "")) {
```
Comment (cpsievert) [inline R/tags.R:503]: Pretty sure this would be faster? Or do we have to be considerate of 

```suggestion
    if (!nzchar(attrib)) {
```

Comment (schloerke) [reply R/tags.R]: Using a combo of `names2(attribs)` with `attrib == ""`

File: R/tags.R [hunk 4]
```diff
@@ -292,7 +292,14 @@ tagFunction <- function(func) {
 #' @export
 tagAppendAttributes <- function(tag, ...) {
   throw_if_tag_function(tag)
-  tag$attribs <- c(tag$attribs, dropNullsOrEmpty(dots_list(...)))
+  newAttribs <- dropNullsOrEmpty(dots_list(...))
+  if (any(names2(newAttribs) == "")) {
```
Comment (cpsievert) [inline R/tags.R:296]: Just to be consistent

```suggestion
  if (any(!nzchar(names2(newAttribs)))) {
```

**Review by cpsievert — APPROVED** [review]

File: NEWS.md [hunk 5]
```diff
@@ -8,6 +8,8 @@
 
 * Closed #197: Fixed rendering of boolean attributes in <script> tags rendered via renderDependencies() (#197, thanks @atusy).
 
+* Closed #222; Unnamed attributes are no longer allowed to be appended via `tagAppendAttribs()`. When trying to print unnamed tag attribs, a better error message is provided. (#229)
```
Comment (cpsievert) [inline NEWS.md:11]:
```suggestion
* Closed #222: Unnamed attributes are no longer allowed to be appended via `tagAppendAttribs()`. When trying to print unnamed tag attribs, a better error message is provided. (#229)
```

**Pull request merged**